services/venice-server/src/main/java/com/linkedin/venice/listener/StorageReadRequestsHandler.java

```diff
@@ -448,7 +448,8 @@ private CompletableFuture<ReadResponse> handleMultiGetRequestInParallel(
           if (responseKeySizeList != null) {
             responseKeySizeList.set(subChunkCur, key.keyBytes.remaining());
           }
-          int subPartitionId = getSubPartitionId(key.partitionId, topic, partitionerConfig, key.keyBytes.array());
+          int subPartitionId =
+              getSubPartitionId(key.partitionId, topic, partitionerConfig, ByteUtils.extractByteArray(key.keyBytes));
```

Comment:
> It can contain extra bytes causing issues for deserialization

For my curiosity, what was the reason for this? Is there a size limit for deserialization?

Reply:
We try to reduce data copy and set the limits for the data in a BB and reuse the underlying `byte[]` where possible

`[0,1,2,3,4,5,...]`: The relevant data items might just be 5 onwards. So, in BB, we'd set position of BB to `5`

`BB.array()` will return the whole array, which is not the intended data